tests/foreman/api/test_puppetmodule.py

```diff
@@ -19,28 +19,30 @@
 
 from robottelo.constants import PUPPET_MODULE_NTP_PUPPETLABS
 from robottelo.helpers import get_data_file
-from robottelo.test import APITestCase
 
 
-class RepositorySearchTestCase(APITestCase):
-    """Tests that search for puppet modules and filter by repository."""
+@pytest.fixture(scope='module')
+def make_product_repo():
+    product = entities.Product().create()
+    repo = entities.Repository(content_type='puppet', product=product).create()
+    return {'product': product, 'repo': repo}
 
-    @classmethod
-    def setUpClass(cls):
-        """Create a product. Make it available as ``cls.product``."""
-        super().setUpClass()
-        cls.product = entities.Product().create()
 
-    def setUp(self):
-        """Create a puppet repository as ``self.repository``.
+@pytest.fixture(scope='function')
+def make_cv(make_product_repo):
+    with open(get_data_file(PUPPET_MODULE_NTP_PUPPETLABS), 'rb') as handle:
+        make_product_repo['repo'].upload_content(files={'content': handle})
```

Comment:
I don't think we need to run upload_content against the repository at a function scope, that should only have to be done once for the repository, at the module or class scope.

The old unittest structure did just that - uploaded to the repository once for the  `ContentViewVersionSearchTestCase` class, and then created a new ContentView entity for each test function individually.

Reply:
Thank you for pointing this out! I have probably overlooked or misunderstood the scoping. Now it should be same flow as the original structure. Please re-review.